napari/__main__.py

```diff
@@ -272,16 +282,23 @@ def _run():
             # if the requested plugin/widget is not available.
             _initialize_plugins()
             plugin_manager.discover_widgets()
-            pname, *wnames = args.with_
-            if wnames:
-                for wname in wnames:
+            for plugin in args.with_:
+                pname, *wnames = plugin
+                if '__all__' in wnames:
+                    for name, (_pname, _wnames) in _npe2.widget_iterator():
+                        if name != 'dock' or pname != _pname:
+                            continue
+                        wnames = _wnames
```

Comment:
```suggestion
                        if name == 'dock' and pname == _pname:
                             wnames = _wnames
                             break
```

Reply:
it's also perhaps worth noting that this will work for npe2 plugins, but not for older plugins (which use [this fallback search here](https://github.com/napari/napari/blob/main/napari/_qt/qt_main_window.py#L679-L682)).  that's ok with me :joy: ... but just a heads up

Reply:
Hmm true, let me think about it :D